src/services/discord-service.ts

```diff
@@ -0,0 +1,16 @@
+import axios from "axios";
+import { UserGuild } from "../models/discord";
+
+export class DiscordService {
+  static async getGuilds(token: string): Promise<UserGuild[] | undefined> {
+    return await axios.request<UserGuild[]>({
+      headers: {
+        Authorization: token
+      },
+      method: 'GET',
+      baseURL: 'https://discord.com/api/v8',
+      url: '/users/@me/guilds'
+    })
+      .then((res) => res != null ? res.data : undefined);
+  }
```

Comment:
res object will never be undefined/null, thus the undefined check does nothing and function will always return `Promise<UserGuild[]>`